Add a Resolving Merge Conflicts section to the README (#56)
open-sauced/pizza-verse: README.md

6. Submit a **pull request** to have your contribution reviewed and considered for merging into the main repository.
7. Engage with other contributors and maintainers through comments and discussions.

## Keeping Branch Up to Date and Resolving Merge Conflicts

> Merge conflicts are something you'll commonly encounter when contributing to an open source project. They usually occur when there are changes on the same line(s) in the same file(s) from two different branches.
> 
> Since the main purpose in this repository is to add a new item to the `pizza-facts-&-trivia.md`, `pizza-recipes.md` and `regional-pizza.md` files, there is a chance that you will encounter merge conflicts that you need to resolve before your pull request (PR) can be merged. It happens when maintainers have merged PRs before yours while you're working on your changes or waiting for your PR to be reviewed.

Reviewer: [suggestion] This explanation is good so far. However, I highly recommend describing merge conflicts in the context of this repository. This helps contributors gain a better understanding of why they happen and how to go about fixing them. 
```suggestion

Since the main purpose in this repository is to add a new item to the `pizza-facts-&-trivia.md`, `pizza-recipes.md` and `regional-pizza.md` files, there is a chance that you will encounter merge conflicts. These are changes that are made on the same line(s) in the same file(s) from two different branches. This is due maintainers merging PRs  while you're working on your own changes or waiting for your PR to be reviewed.
```

Author: I understand that we want to explain merge conflicts in the context of this repository. And that's what I do in the 2nd paragraph.

I think giving a general context of why conflicts happen in the beginning can give folks the knowledge of it before we walk them through why it happens in this repo.

What do you think?

Reviewer: I get that @adiati98 .I was just consider that the paragraphs would sound repetitive. Hence why I came up with the suggestion you see in my in-line comment.

Author: Ah, I get that @CBID2. Let me see how I can combine both of our purposes. :)

Author: How about this, @CBID2?

Merge conflicts occur when there are changes on the same line(s) in the same file(s) from two different branches. Since the main purpose in this repository is to add a new item to the `pizza-facts-&-trivia.md`, `pizza-recipes.md`, and `regional-pizza.md` files, there is a chance that you will encounter and need to resolve conflicts because maintainers might have merged PRs before yours while you're working on your changes or waiting for your PR to be reviewed.

Reviewer: Perfect @adiati98! 😊

First, you must update your forked (`origin`) repository:

1. Go to your forked repository on GitHub.
2. Click the "Sync fork" button.
3. Click the green "Update branch" button.

Then, pull the latest changes in the `main` branch in the `origin` repository to your local working branch by following these steps in your terminal:

1. Go to your working branch.

   ```bash
   git checkout <your-branch-name>
   ```

2. Pull the latest changes with this command:

   ```bash
   git pull origin main
   ```

### Resolving Merge Conflicts

First, you need to pay attention to the conflicts. On which line(s) does the conflict happen? What are the differences?

Everything between the `<<<<<<< HEAD` and `=======` is the changes that you worked on (current changes). And everything between the `=======` to `>>>>>>>` is the incoming changes from the remote `main` branch that you have pulled.

Now, you must decide how you want to resolve the conflicts. Because the contributions in this repository are to add new items, you want to keep both yours and the incoming changes, which are the items from previous contributor(s).

Follow these steps to resolve the merge conflicts in this repository:

> 1. Click the "Accept Both Change" option on the top of your workspace in VSCode.
> 2. Fix anything necessary, such as duplicate sentences, etc. **Tips**: If you're confused, look at the markdown file in this repository to compare the current state of the content with your local file when fixing.
> 3. Move your item to the end of the list.
> 4. Add and commit your changes.

Reviewer: [suggestion] Some contributors might want to know more about this feature or not have it setup, so I highly recommend linking an article to help them with this. 
```suggestion
1. Click the "Accept Both Change" option on the top of your workspace in VSCode. If you don't have this feature setup already, check out [this article by Lee Stanton](https://www.alphr.com/vs-code-open-merge-editor/) to help you. 
2. Fix anything necessary, such as duplicate sentences, etc. **Tips**: If you're confused, look at the markdown file in this repository to compare the current state of the content with your local file when fixing.
3. Move your item to the end of the list.
4. Add and commit your changes.
```

Author: Oh, good to know. Because I think I never have it setup but I somehow have it 😅.
